websub-generic-integration-tests/tests/test_content_negotiation.bal

```diff
@@ -0,0 +1,167 @@
+        http:Response response = new;
+        string media_type = req.getHeader("Accept");
+        string language_type = req.getHeader("Accept-Language");
+        if (media_type == "application/json" && language_type == "de-DE") {
+            websub:addWebSubLinkHeader(response, [WebSubHub.subscriptionUrl], WEBSUB_TOPIC_ONE);
+            response.statusCode = 202;
+            var result = caller->respond(response);
+            if (result is error) {
+                log:printError("Error responding ", err = result);
+            }
+        } else {
+            response.statusCode = 406;
+            var res = caller->respond(response);
+        }
+    }
+
+    //Publisher need accept header only and agree with application/json as Accept header
+    resource function get discoveryAcceptHeaderOnly(http:Caller caller, http:Request req) {
+        http:Response response = new;
+        string media_type = req.getHeader("Accept");
+        if (media_type == "application/json") {
+            websub:addWebSubLinkHeader(response, [WebSubHub.subscriptionUrl], WEBSUB_TOPIC_ONE);
+            response.statusCode = 202;
+            var result = caller->respond(response);
+            if (result is error) {
+                log:printError("Error responding ", err = result);
+            }
+        } else {
+            response.statusCode = 406;
+            var res = caller->respond(response);
+        }
+    }
+
+    //Publisher need accept language header only and agree with de-DE as Accept-Language header
+    resource function get discoveryAcceptLanguageOnly(http:Caller caller, http:Request req) {
+        http:Response response = new;
+        string language_type = req.getHeader("Accept-Language");
+        if (language_type == "de-DE") {
+            websub:addWebSubLinkHeader(response, [WebSubHub.subscriptionUrl], WEBSUB_TOPIC_ONE);
+            response.statusCode = 202;
+            var result = caller->respond(response);
+            if (result is error) {
+                log:printError("Error responding ", err = result);
+            }
+        } else {
+            response.statusCode = 406;
+            var res = caller->respond(response);
+        }
+    }
+}
+
+@test:Config{}
+//SubscriberServiceConfig's accept and acceptLanguage field values match with publisher's acceptable Accept and Accept-Language header values.
+function testMatchAcceptAndAcceptLanguage() {
+    http:Client clientEndpoint = new("http://localhost:24080");
+    http:Request req = new;
+    req.setHeader("Accept", "application/json");
+    req.setHeader("Accept-Language", "de-DE");
+    var result = clientEndpoint->get("/publisherService/discoveryWithAcceptAndAcceptLanguage", req);
+    HttpResponseDetails responseDetails = fetchHttpResponse(result);
+    test:assertEquals(responseDetails.statusCode, http:STATUS_ACCEPTED, msg = "unsupported content type or media type.");
```

Comment:
The message is used to indicate what went wrong when a test fails, so I think we should rephrase these to something like
```suggestion
    test:assertEquals(responseDetails.statusCode, http:STATUS_ACCEPTED, msg = "expected the publisher to support the content type and media type");
```